operator/flags.go

```diff
@@ -138,9 +115,74 @@ func init() {
 	flags.Bool(operatorOption.EnableMetrics, false, "Enable Prometheus metrics")
 	option.BindEnv(operatorOption.EnableMetrics)
 
-	flags.String(option.IPAM, ipamOption.IPAMHostScopeLegacy, "Backend to use for IPAM")
+	var defaultIPAM string
+	switch binaryName {
+	case "cilium-operator":
+		defaultIPAM = ipamOption.IPAMHostScopeLegacy
+	case "cilium-operator-aws":
+		defaultIPAM = ipamOption.IPAMENI
+	case "cilium-operator-azure":
+		defaultIPAM = ipamOption.IPAMAzure
+	case "cilium-operator-generic":
+		defaultIPAM = ipamOption.IPAMOperator
+	}
+
+	flags.String(option.IPAM, defaultIPAM, "Backend to use for IPAM")
 	option.BindEnv(option.IPAM)
 
+	rootCmd.PreRunE = func(cmd *cobra.Command, args []string) error {
```

Comment:
It's not particularly pretty, but seems okay to me. A better options would be to have a wrapper binary that calls out to the correct implementation, but I think this will do to start with.